### PR TITLE
added errWriter so getting a shelfWriter no longer returns an error

### PR DESCRIPTION
**Pull request by woutslakhorst**
Got a bit tired with all the error checking when getting shelves....

### CODE REVIEW

File: store.go [hunk 1]
```diff
@@ -276,3 +276,40 @@ func (n NilReader) Stats() ShelfStats {
 		ShelfSize:  0,
 	}
 }
+
+// NewErrorWriter returns a Writer that will return the error for every method
+func NewErrorWriter(err error) Writer {
+	return errWriter{err: err}
```
Comment (gerardsn) [inline store.go:282]: The error should probably be wrapped in `ErrDatabase`?

File: store.go [hunk 2]
```diff
@@ -241,8 +241,8 @@ func OnRollback(fn func()) TxOption {
 type WriteTx interface {
 	ReadTx
 	// GetShelfWriter returns the specified shelf for writing. If it doesn't exist, it will be created.
-	// Returns a ErrDatabase if unsuccessful.
-	GetShelfWriter(shelfName string) (Writer, error)
+	// Returns a ErrWriter if unsuccessful.
```
Comment (gerardsn) [inline store.go:244]: `errWriter` is private

File: store.go [hunk 3]
```diff
@@ -276,3 +276,40 @@ func (n NilReader) Stats() ShelfStats {
 		ShelfSize:  0,
 	}
 }
+
+// NewErrorWriter returns a Writer that will return the error for every method
+func NewErrorWriter(err error) Writer {
+	return errWriter{err: err}
+}
+
+// ErrWriter is a shelfWriter that already failed, but to reduce the error count it'll fail on the write operation
```
Comment (gerardsn) [inline store.go:285]:
```suggestion
// errWriter is a shelfWriter that already failed, but to reduce the error count it'll fail on the write operation
```

**Review by gerardsn — CHANGES_REQUESTED** [review]
Just noting that we now have a`NilReader` and `errWriter` that are specific to bbolt

**Review by reinkrul — CHANGES_REQUESTED** [review]

**Review by gerardsn — APPROVED** [review]

**Comment by gerardsn** [comment]
Maybe add test to confirm that returned errors are `ErrDatabase` errors

**Pull request merged**